File: mobile/lib/models/exercise.dart
import 'dart:convert';

class Exercise {
  final String id;
  final String name;
  final String bodyPart;
  final String category;
  final String? description;
  final String? imageUrl;

  Exercise({
    required this.id,
    required this.name,
    required this.bodyPart,
    required this.category,
    this.description,
    this.imageUrl,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['_id'],
      name: json['name'],
      bodyPart: json['bodyPart'],
      category: json['category'],
      description: json['description'],
      imageUrl: json['imageUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'bodyPart': bodyPart,
      'category': category,
      'description': description,
      'imageUrl': imageUrl,
    };
  }
}

File: backend/models/Exercise.js
import mongoose from "mongoose";

const exerciseSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
    unique: true,
  },
  bodyPart: {
    type: String,
    required: true,
    enum: ["chest", "back", "legs", "shoulders", "arms", "core"],
  },
  category: {
    type: String,
    enum: [
      "barbell",
      "dumbbell",
      "machine",
      "bodyweight",
      "cable",
      "kettlebell",
      "bands",
      "other",
    ],
    default: "other",
  },
  defaultReps: {
    type: Number,
    default: 10,
    min: 1,
  },
  description: {
    type: String,
    default: "",
  },
  imageUrl: {
    type: String,
    default: "",
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

// 添加索引以提升查詢效能
exerciseSchema.index({ name: 1 });
exerciseSchema.index({ bodyPart: 1 });
exerciseSchema.index({ category: 1 });

export default mongoose.model("Exercise", exerciseSchema);
